File: app/src/main/java/com/absinthe/anywhere_/services/CollectorTileService.java
package com.absinthe.anywhere_.services;

import android.content.Intent;
import android.os.Build;
import android.service.quicksettings.Tile;
import android.service.quicksettings.TileService;

import androidx.annotation.RequiresApi;

import com.absinthe.anywhere_.R;
import com.absinthe.anywhere_.utils.PermissionUtils;

@RequiresApi(api = Build.VERSION_CODES.N)
public class CollectorTileService extends TileService {
    private static CollectorTileService sInstance;

    public CollectorTileService() {
        sInstance = this;
    }

    @Override
    public void onClick() {
        Tile tile = getQsTile(); // Get Tile
        if (tile.getState() == Tile.STATE_ACTIVE) {
            CollectorService.closeCollector(this);

            tile.setState(Tile.STATE_INACTIVE);
            tile.setLabel(getString(R.string.tile_collector_on));
        } else {
            if (PermissionUtils.checkOverlayPermission(this)) {
                CollectorService.startCollector(this);
            }

            tile.setState(Tile.STATE_ACTIVE);
            tile.setLabel(getString(R.string.tile_collector_off));
        }

        sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
        tile.updateTile();
    }

    public static CollectorTileService getInstance() {
        return sInstance;
    }
}

File: app/src/main/java/com/absinthe/anywhere_/services/CollectorTileService.kt
package com.absinthe.anywhere_.services

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.content.ServiceConnection
import android.os.Build
import android.os.IBinder
import android.service.quicksettings.Tile
import android.service.quicksettings.TileService
import androidx.annotation.RequiresApi
import com.absinthe.anywhere_.R

@RequiresApi(api = Build.VERSION_CODES.N)
class CollectorTileService : TileService() {

    private var isBound = false
    private var collectorService: CollectorService? = null
    private val conn = object : ServiceConnection {
        override fun onServiceDisconnected(name: ComponentName?) {
            isBound = false
        }

        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
            isBound = true
            collectorService = (service as CollectorService.CollectorBinder).service
            collectorService?.startCollector()
        }

    }

    override fun onClick() {
        qsTile?.let {
            if (isBound) {
                collectorService?.startCollector()
            } else {
                bindService(Intent(this, CollectorService::class.java), conn, Context.BIND_AUTO_CREATE)
            }

            it.state = Tile.STATE_ACTIVE
            it.label = getString(R.string.tile_collector_off)
            sendBroadcast(Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS))
            it.updateTile()
        }
    }
}